doc/classes/TileData.xml

```diff
@@ -199,7 +199,8 @@
 		</member>
 		<member name="flip_v" type="bool" setter="set_flip_v" getter="get_flip_v" default="false">
 		</member>
-		<member name="material" type="ShaderMaterial" setter="set_material" getter="get_material">
+		<member name="material" type="Material" setter="set_material" getter="get_material">
+			The [Material] to use for this [TileData].
```

Comment:
Might be worth mentioning supported types (ShaderMaterial, CanvasItemMaterial) here.